lib/models/obms.js

```diff
@@ -0,0 +1,183 @@
+// Copyrighr 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = ObmsModelFactory;
+
+ObmsModelFactory.$provide = 'Models.Obms';
+ObmsModelFactory.$inject = [
+    'Model',
+    'Services.Waterline',
+    '_',
+    'Services.Encryption',
+    'Constants',
+    'Promise',
+    'util',
+    'Errors',
+    'Assert'
+];
+
+function ObmsModelFactory (
+    Model,
+    waterline,
+    _,
+    encryption,
+    Constants,
+    Promise,
+    util,
+    Errors,
+    assert
+) {
+    var secretsPattern = _.reduce(Constants.Logging.Redactions, function(pattern, regex) {
```

Comment:
@yyscamper Redactions would actually be easier to aggregate if they were strings, not patterns, so I'm all for changing this.  I don't really think there's a use case for taking them individually, so I'd vote for structuring them so that they are easy to merge.

As for a common redaction service, I think this is unnecessary if encryption is managed in the models.  My preference is to make each model responsible for encrypting any secrets it might contain since models are responsible for their own content.  Encryption is a form of redaction since plain text secrets are replaced with a cipher.  I also think that `toJSON()` should remove any secret properties from the model before it is used in logs, views, etc since redacted data has no value to the user.